presto-main/src/main/java/com/facebook/presto/operator/Driver.java

```diff
@@ -401,6 +405,27 @@ private void processNewSources()
                 }
             }
 
+            for (int index = operators.size() - 1; index >= 0; index--) {
+                if (operators.get(index).isFinished()) {
+                    // close and remove all operators up to and including this operator
```

Comment:
"Up to" is kind of ambiguous here. We generally write out operators from bottom to top in Presto. And imagine that's where "up to" here comes from. However, the operators that got closed here are the upstream operators.